web-app/web-frontend/script.js

```diff
@@ -1,193 +1,224 @@
 // NOTE: this example uses the chess.js library:
 // https://github.com/jhlywa/chess.js
 
-var board = null
-var game = new Chess()
-game.load("rnbqkbnr/ppp1pppp/8/4PP2/8/P7/1PpP2PP/RNBQKBNR b KQkq - 0 5");
-var $status = $('#status')
-var $pgn = $('#pgn')
-var $gan = $('#gan')
-
-function onDragStart (source, piece, position, orientation) {
+let board = null;
+const game = new Chess();
+const $status = $('#status');
+const $pgn = $('#pgn');
+const $gan = $('#gan');
+
+/** checks if game is over if valid player move
+ * @param {Object} source location
+ * @param {Object} piece current selection
+ * @param {Object} position location
+ * @param {Object} orientation location
+ * @return {boolean} valid move
+*/
+function onDragStart(source, piece, position, orientation) {
   // do not pick up pieces if the game is over
-  if (game.game_over()) return false
+  if (game.game_over()) return false;
 
   // only pick up pieces for the side to move
   if ((game.turn() === 'w' && piece.search(/^b/) !== -1) ||
       (game.turn() === 'b' && piece.search(/^w/) !== -1)) {
-    return false
+    return false;
   }
 }
 
-
-function onDrop (source, target) {
+/** determines if move is promotion or valid and makes moves
+ * @param {Object} source move from source location
+ * @param {Object} target move to target location
+ * @return {boolean} valid move
+*/
+function onDrop(source, target) {
   // see if the move is a promotion
-  if (is_promotion({
+  if (isPromotion({
     game: game,
-    move: {from: source, to: target}
-  }) ) {  
-   
-    processUserInput(source, target)
-    //doesn't work b/c on drop there is no value for promotion so only changes when you drop again, must find a way to move the piece when there is no drop 
-   /*choose_piece({
-        onchosen: function(idPost) {
-              game.move({ from: source, to: target, promotion: idPost });
-         }
-   }*/
-   }
-   else{ // normal move
+    move: {from: source, to: target},
+  })) {
+    processUserInput(source, target);
+  } else { // normal move
     var move = game.move({
       from: source,
-      to: target
-    })
-   }
-
+      to: target,
+    });
+  }
   // illegal move
-  if (move === null) return 'snapback'
-
-  updateStatus()
+  if (move === null) return 'snapback';
+  updateStatus();
```

Comment:
Having a line between makes it easier to understand that `updateStatus` isn't supposed to be part of the if statement